### PR TITLE
[DOCS] Document disallowed explicit conversions between unrelated contract types.

### PR DESCRIPTION
**Pull request by ekpyron**
Came up on gitter.
I just quickly wrote something up to track this - we may want to reformulate and maybe we want to include this elsewhere as well or I missed it being mentioned elsewhere already...

### CODE REVIEW

**Review by erak — DISMISSED** [review]

File: docs/050-breaking-changes.rst [hunk 1]
```diff
@@ -101,6 +101,13 @@ For most of the topics the compiler will provide suggestions.
   ``c.transfer(...)`` to ``address(c).transfer(...)``,
   and ``c.balance`` to ``address(c).balance``.
 
+* Explicit conversions between arbitrary contract types that are not part of the same
```
Comment (chriseth) [inline docs/050-breaking-changes.rst]: You cannot convert siblings between each other either, so they have to inherit or be inherited from.

File: docs/050-breaking-changes.rst [hunk 2]
```diff
@@ -101,6 +101,13 @@ For most of the topics the compiler will provide suggestions.
...
+  inheritance hierarchy are now disallowed. Instead, you need to explicitly convert
```
Comment (chriseth) [inline docs/050-breaking-changes.rst]: Perhaps better: `If you are sure the contract is of the type you want to convert to, you can work around this by converting to address first`

**Review by chriseth — DISMISSED** [review]

File: docs/050-breaking-changes.rst (hunk 1 above)
Comment (ekpyron) [reply docs/050-breaking-changes.rst]: Actually we even only allow it from a contract type to one of its ancestor types, not the other way round - I'll change it.

File: docs/050-breaking-changes.rst (hunk 2 above)
Comment (ekpyron) [reply docs/050-breaking-changes.rst]: Actually ``If you are sure that the contract is compatible with the contract type you want to convert to`` should be enough, resp. clearer, right?

**Review by chriseth — APPROVED** [review]

**Pull request merged**